python-avd/pyavd/_eos_designs/structured_config/underlay/vlans.py

```diff
@@ -53,13 +52,13 @@ def _update_underlay_vlan_trunk_groups(self: AvdStructuredConfigUnderlayProtocol
         if self.inputs.enable_trunk_groups is not True:
             return
 
-        for peer in self._avd_peers:
-            peer_facts = self.shared_utils.get_peer_facts(peer, required=True)
-            for uplink in peer_facts["uplinks"]:
-                if uplink["peer"] == self.shared_utils.hostname:
-                    if (peer_trunk_groups := get(uplink, "peer_trunk_groups")) is None:
+        for peer in self.facts.downlink_switches:
+            peer_facts = self.shared_utils.get_peer_facts(peer)
+            for uplink in peer_facts.uplinks:
+                if uplink.peer == self.shared_utils.hostname:
```

Comment:
Could revert the logic to avoid nesting ifs
```suggestion
                if uplink.peer != self.shared_utils.hostname:
                    continue
                ...
```

Reply:
done